templates/recipe.j2

```diff
@@ -0,0 +1,12 @@
+{%- for module in modules_ext.modules %}
+  {%- if let Some(from_file) = module.from_file %}
+{{ ["from_file",from_file]|join(": ") }}
+    {%- set from_file_module = self::get_module_from_file(from_file) %}
+    {%- for item in from_file_module.modules %}
+{{ item|yaml|indent(2) }}
+    {%- endfor %}
+  {%- else %}
+{{ module|yaml|indent(2) }}
+  {%- endif %}
+{%- endfor %}
```

Comment:
I've noticed some misalignment in the resulting yaml indents that is out of spec. You could write a function that flattens the module imports and just use serde to print out the file, that way the file follows the spec always and we don't have to update a template if we make any changes to the structure of the recipe.